libcrux-sha3/tests/sha3.rs

```diff
@@ -1,3 +1,343 @@
+        let mut state = incremental::shake256_init();
+        incremental::shake256_absorb_final(&mut state, test_vectors::HELLO);
+
+        let mut digest = [0u8; 136];
+        incremental::shake256_squeeze_first_block(&mut state, &mut digest);
+        assert_eq!(
+            hex::encode(&digest),
+            &test_vectors::shake256::HELLO_FIVE_BLOCKS[..272]
+        );
+
+        // Test squeezing next block (136 bytes)
+        incremental::shake256_squeeze_next_block(&mut state, &mut digest);
+        assert_eq!(
+            hex::encode(&digest),
+            &test_vectors::shake256::HELLO_FIVE_BLOCKS[272..544]
+        );
+
+        // Test squeezing 3 blocks (408 bytes = 3 * 136)
+        state = incremental::shake256_init();
+        incremental::shake256_absorb_final(&mut state, test_vectors::HELLO);
+
+        let mut digest = [0u8; 408];
+        incremental::shake256_squeeze_first_three_blocks(&mut state, &mut digest);
+        assert_eq!(
+            hex::encode(&digest),
+            &test_vectors::shake256::HELLO_FIVE_BLOCKS[..816]
+        );
+
+        // Test squeezing 5 blocks (680 bytes = 5 * 136)
+        state = incremental::shake256_init();
+        incremental::shake256_absorb_final(&mut state, test_vectors::HELLO);
+
+        let mut digest = [0u8; 680];
+        incremental::shake256_squeeze_first_five_blocks(&mut state, &mut digest);
+        assert_eq!(
+            hex::encode(&digest),
+            &test_vectors::shake256::HELLO_FIVE_BLOCKS[..1360]
+        );
+    }
+
+    #[test]
+    fn sha3_shake128_absorb() {
+        let mut state = incremental::Shake128Xof::new();
+        state.absorb_final(b"Hello, ");
+
+        let mut digest = [0u8; 32];
+        state.squeeze(&mut digest);
+        let expected = "62dac7f538d3c56e66a1e0ccda69f4b6c8f6269572ad9312c7a04a2228b474a5";
+        assert_eq!(hex::encode(&digest), expected);
+
+        // ---
+
+        state = incremental::Shake128Xof::new();
+        state.absorb(b"Hello, ");
+        state.absorb_final(b"World!");
+
+        state.squeeze(&mut digest);
+        assert_eq!(
+            hex::encode(&digest),
+            &test_vectors::shake128::HELLO_FIVE_BLOCKS[..64]
+        );
+    }
+
+    #[test]
+    fn sha3_shake256_absorb() {
+        let mut state = incremental::Shake256Xof::new();
+        state.absorb_final(b"Hello, ");
+
+        let mut digest = [0u8; 32];
+        state.squeeze(&mut digest);
+        let expected = "018680a686f24f889fe4613dba0058ea1b035b7270a8c26b363f42557bbd991a";
+        assert_eq!(hex::encode(&digest), expected);
+
+        // ---
+
+        state = incremental::Shake256Xof::new();
+        state.absorb(b"Hello, ");
+        state.absorb_final(b"World!");
+
+        state.squeeze(&mut digest);
+        assert_eq!(
+            hex::encode(&digest),
+            &test_vectors::shake256::HELLO_FIVE_BLOCKS[..64]
+        );
+    }
+}
+
 #[test]
 fn sha3_kat_oneshot() {
```

Comment:
As you noted already, the two tests down here do the same.
Since you added more, and more systematically, you could drop both of them.